web/src/features/charts/elements/AreaGraphLayers.tsx

```diff
@@ -75,8 +75,8 @@ function AreaGraphLayers({
         // Therefore, we copy all datapoints and make sure
         // both a start and an end are present to ensure
         // proper display of missing points
-        const datapoints = [
-          ...layer.datapoints.map((d: { data: AreaGraphElement }) => [
+        const datapoints: any[] = layer.datapoints.flatMap(
```

Comment:
you like the explicity any better?

Reply:
Should work but I have to do some testing, I won't have time to do that today though.

Reply:
If we remove the any[] cast here I think we can merge this.

Reply:
Right now the any[] cast gets a warning. which i thought was ok since there are lots of those warnings. but if you remove it then you get an error below. i tried to type it better but i don't really understand what's happening here, even after reading the comment.

Reply:
It's a pretty complex type as it represent all the data for the graphs. Let's stick with the any cast for now. (Even though the CI will complain 😅)

Reply:
Just ran it again locally and I can see no errors? Where are you seeing the error?

Reply:
if i take the any cast out i see this: 
<img width="806" alt="image" src="https://github.com/electricitymaps/electricitymaps-contrib/assets/20405975/eaac8f61-22cb-4098-97f3-79762f935ba5">

Reply:
Let's keep it like this then, and hopefully we can get a better typing in the future.